### ISSUE TITLE
"Many" rays may not intersect the optic

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/geometric-optics#289]
This is a relatively serious physics bug.

In Many rays mode, it's possible that no rays may intesect the optic, and therefore no image is formed.  This can happen by reducing the optic diameter, or by zooming out and moving the Object further away from the optic.  I can happen in both screens, and some screenshots are shown below. 

@arouinfar pointed out that this was not a problem way back in [1.0.0-dev.5](https://phet-dev.colorado.edu/html/geometric-optics/1.0.0-dev.5/phet/geometric-optics_en_phet.html), which has different behavior for Many rays.

<img width="1040" alt="screenshot_1512" src="https://user-images.githubusercontent.com/3046552/149027355-4446882b-3ea7-4b22-9353-270df26442f9.png">

<img width="1045" alt="screenshot_1511" src="https://user-images.githubusercontent.com/3046552/149027202-0c9237c5-9673-4dbe-a95f-4bfe61aad217.png">

Having only 1 ray intersect the optic is also problematic, because an image is supposed to be formed at the intersection of refracted/reflected rays.  For example, see the screenshot below.  So at least 2 rays need to intersect the optic.

<img width="1039" alt="screenshot_1513" src="https://user-images.githubusercontent.com/3046552/149027531-9b198961-bdb7-4e73-8f8a-5d19ddb748cf.png">

**Comment by pixelzoom** [comment]
> @arouinfar pointed out that this was not a problem way back in [1.0.0-dev.5](https://phet-dev.colorado.edu/html/geometric-optics/1.0.0-dev.5/phet/geometric-optics_en_phet.html), which has different behavior for Many rays.

In 1.0.0-dev.5, all of the rays intersect the optic. I don't know the history of why that was changed.  Searching for "Many" in closed GitHub issues doesn't find anything relevant.

I don't know how to address this problem while making Many rays look like it does now, AND make are least 2 rays always intersect the optic.

@arouinfar thoughts?

**Comment by ariel-phet** [comment]
@pixelzoom given that the "many rays" is a representation (there are in a sense nearly infinite rays of which we are showing a subset) I think it would be fine to "add" rays as necessary to ensure you are always hitting the optic. So perhaps some simple algorithm that reduces the angle between rays as you move the object further from the optic and adds rays as necessary to keep the total angular spread of the many rays reasonably consistent?

**Comment by pixelzoom** [comment]
At 1/22/2022 design meeting, we decided to:

(1) Change the optic diameter range to [60,120] for both screens.
(2) Increase the number of rays in "steps", as the Object gets further from the optic. (The step size is currently 300 cm.)

After I made those changes, a bit more discussion was needed with @arouinfar and @ariel-phet.  Even with more rays, there were cases where the Object was to the left of the optic, but no rays reached the optic, and thus no Image was formed.  This occurred mostly when the Object was moved an extreme distance from the optical axis, when zoomed out.  

So we decided to make 3 additional changes:

(3) Constrain the "model bounds" (which constrains dragging) to 100 cm above and below the optical axis. Run with `?debugModelBounds` to see the model bounds rendered as a red rectangle.  Note that these bounds constrain dragging of all things -- Object, Light Sources, and Projection Screen.
(4) Increase the "fan angle" of the Many rays from 90 degrees to 120 degrees.
(5) Increase the minimum number of Many rays from 15 to 20. (This is the number of rays when fully zoomed in.) 

@arouinfar ready for review.
